### PR TITLE
Remove collateralized_contract.ts and add test scenarios for collateralizer

### PR DESCRIPTION
**Pull request by graemecode**
Part of splitting up the collateralizer unit test PR.

### CODE REVIEW

File: test/ts/unit/collateralizer/scenarios/successful_return.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                expectedRepaymentValue: Units.ether(0.5),
+            },
+            {
+                timestamp: moment()
+                    .subtract(7, "days")
+                    .unix(),
+                expectedRepaymentValue: Units.ether(1),
+            },
+        ],
+        termEndTimestamp: 0,
```
Comment (nadavhollander) [inline test/ts/unit/collateralizer/scenarios/successful_return.ts]: This very well may be an issue I introduced originally, but I don't understand why this `termEndTimestamp` is 0.  Seems like it should be a timestamp after the last timestamp in `expectedRepaymentValueSchedule` and before now.

File: test/ts/unit/collateralizer/scenarios/successful_return.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        valueRepaidToDate: Units.ether(3),
+        expectedRepaymentValueSchedule: [
+            {
+                timestamp: moment()
+                    .subtract(7, "days")
+                    .unix(),
+                expectedRepaymentValue: Units.ether(2),
+            },
+        ],
+        termEndTimestamp: moment()
```
Comment (nadavhollander) [inline test/ts/unit/collateralizer/scenarios/successful_return.ts]: Ditto with this timestamp -- shouldn't this be after the last expected repayment?

File: test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                    .unix(),
+                expectedRepaymentValue: Units.ether(0.5),
+            },
+        ],
+        agreementId: web3.sha3("Arbitrary 32 byte id for unsuccessful seizure scenario #5"),
+    },
+    {
+        description: "(Grace Period = 90 Days) Debt is not currently in state of default",
+        ...defaultArgs,
+        gracePeriodInDays: new BigNumber(90),
```
Comment (nadavhollander) [inline test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts]: Again, probably my mistake, but this needs to specify `valueRepaidToDate` as being not in a state of default.  The default args specify the value repaid as being insufficient.

File: test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        agreementId: web3.sha3("Arbitrary 32 byte id for unsuccessful seizure scenario #6"),
+    },
+    {
+        description:
+            "(Grace Period = 90 Days) Debt is currently in default, but grace period has not elapsed",
+        ...defaultArgs,
+        gracePeriodInDays: new BigNumber(90),
+        valueRepaidToDate: Units.ether(0.5).minus(1),
```
Comment (nadavhollander) [inline test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts]: Redundant given the defaults, please remove.

File: test/ts/unit/collateralizer/scenarios/successful_return.ts (hunk 1 above)
Comment (graemecode) [reply test/ts/unit/collateralizer/scenarios/successful_return.ts]: It's actually just mocked that way in MockCollateralizedTermsContract!

**Comment by graemecode** [comment]
♻️  I think this addresses the comments

File: test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts (hunk 3 above)
Comment (nadavhollander) [reply test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts]: It doesn't look like this is fixed?

Comment (graemecode) [reply test/ts/unit/collateralizer/scenarios/unsuccessful_seizure.ts]: whoops, I had been trying to access the expected repayment value directly from defaultArgs, but hadn't committed it properly. Updated to `Units.ether(0.5)` - is that correct?

**Pull request merged**